src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts

```diff
@@ -112,6 +112,27 @@ export function normalizeClientRequestArgs(
     url = args[0]
     logger.info('first argument is a URL:', url)
 
+    // Check if the second provided argument is RequestOptions.
+    // If it is, check if "options.path" was set and rewrite it
+    // on the input URL.
+    // Do this before resolving options from the URL below
+    // to prevent query string from being duplicated in the path.
+    if (
```

Comment:
I've added this check to only respect `options.path` if the second argument to the request is request options. Previously, we checked this on _resolved_ options, which meant options that have already inherited properties from the `URL` instance, potentially.